src/lib/tabwidget/tabwidget.cpp

```diff
@@ -787,8 +787,17 @@ QByteArray TabWidget::saveState()
         if (!webTab)
             continue;
 
-        WebTab::SavedTab tab(webTab);
-        tabList.append(tab);
+        // save state does not support saving restore tab
```

Comment:
Yeah, I don't really like it.
Please remove this check for this patch and let's implement it correctly in followup commit - eg. skip the restore tab (or any other specified URLs) when saving the session properly.